Add section promo

ArtemG22 opened the pull request:
http://restaurant10012023-2.surge.sh/

ArtemG22 committed:
feat: create a PROMO section. Add image, html and css propertiescreate a PROMO section. Add image, html and css properties

ArtemG22 committed:
refactor: change structure and rename images

ArtemG22 committed:
refactor: remove unnecessary indents in properties

ArtemG22 committed:
docs: hide a foreign document

ArtemG22 committed:
refactor: delete extraneous images

ArtemG22 committed:
docs: delete extraneous text files

ArtemG22 committed:
refactor: remove unnecessary elements

ArtemG22 committed:
refactor: delete imported icons

ArtemG22 committed:
feat: add social network icons

ynworld commented on index.html:
```diff
@@ -4,11 +4,54 @@
+                </div>
+                  <div class="buttons-bar">
+                    <div class="buttons">
+                      <a href="#" class="btn btn--full">BOOK A TABLE</a>
```
Let's use CSS to make text uppercase.

ArtemG22 replied:
Edited. Added the necessary CSS properties

ArtemG22 committed:
refactor: format button contents

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,180 @@
+  justify-content: space-around;      
+  padding: 2rem 0 1rem 1rem;    
+  background: linear-gradient(to bottom, black -100%, transparent 90%);
+  height: 8rem;
```
Height seems redundant

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,180 @@
+  display: flex;
+  width: 100%;        
+  justify-content: space-around;      
+  padding: 2rem 0 1rem 1rem;    
```
Looks like only top padding makes sense here. Or I'd make the container `space-between` and then side paddings will work 
<img width="1428" alt="image" src="https://user-images.githubusercontent.com/34685713/194755305-1f2d7680-64de-4ce6-90d1-919080d61041.png">

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,180 @@
+  display: flex;
+  gap: 0.6rem;
+  margin-left: 82%;
+  margin-top: -1.2rem;    
```
Negative margin is a bad practice

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,180 @@
+  list-style: none;
+  display: flex;
+  gap: 0.6rem;
+  margin-left: 82%;
```
It looks like there's something wrong with the layout. Mockup is fixed, percentage should not be applied here

ArtemG22 replied:
fixed

magicsd commented on index.html:
```diff
@@ -4,11 +4,54 @@
-    
+
+    <header class="header">
+      <img class="logo" alt="LAMBDA" src="gallery/logo/lambda-logo.png">
```
Logo is expected to be svg or plain text

ArtemG22 replied:
fixed

magicsd commented on index.html:
```diff
@@ -4,11 +4,54 @@
+            <ul class="social-links">
+              <li><a class="social-link" href="#"><img src="gallery/icon/twitter.png" name="logo-twitter"></a></li>
+              <li><a class="social-link" href="#"><img src="gallery/icon/youtube.png" name="logo-youtube"></a></li>
+              <li><a class="social-link" href="#"><img src="gallery/icon/facebook.png" name="logo-facebook"></a></li>            
```
Icons are expected to be SVG

magicsd commented on index.html:
```diff
@@ -4,11 +4,54 @@
+            <li><a class="main-nav-link" href="#">Reviews</a></li>
+            <li><a class="main-nav-link" href="#">Reservations</a></li>
+          </ul>
+            <ul class="social-links">
```
Same level, but different indents

ArtemG22 replied:
fixed

magicsd commented on index.html:
```diff
@@ -4,11 +4,54 @@
+
+      <main>
+
+        <section class="promo-section">
```
```suggestion
      <main>
        <section class="promo-section">
```

ArtemG22 replied:
fixed

magicsd commented on index.html:
```diff
@@ -4,11 +4,54 @@
+        <section class="promo-section">
+          <div class="promo-banner">
+            <div class="promo-slogan"> 
+              <p class="slogan">the right ingredients<br>for the right food</p>
```
Use CSS to make text lowercased
```suggestion
              <p class="slogan">The right ingredients<br>for the right food</p>
```

ArtemG22 replied:
fixed

magicsd commented on index.html:
```diff
@@ -4,11 +4,54 @@
+                  <div class="buttons-bar">
+                    <div class="buttons">
+                      <a href="#" class="btn btn--full">book a table</a>
+                      <a href="#" class="btn btn--outline">see the menu</a>
```
```suggestion
                      <a href="#" class="btn btn--full">Book a table</a>
                      <a href="#" class="btn btn--outline">See the menu</a>
```

ArtemG22 replied:
fixed

magicsd commented on index.html:
```diff
@@ -4,11 +4,54 @@
+                <div class="promo-decoration">
+                  <img src="gallery/decoration/decoration-1.png" class="decoration" alt="decoration-promo">
+                </div>
+                  <div class="buttons-bar">
```
Fix all the weird indentations please

ArtemG22 replied:
fixed

ArtemG22 committed:
refactor: exclude unnecessary margins in the site header

magicsd commented on css/style.css:
```diff
@@ -16,28 +16,33 @@ html {
-  padding: 2rem 0 1rem 1rem;    
+  width: 100%;
+  justify-content: space-around;
+  padding: 2rem 0 0 0;
```
```suggestion
  padding-top: 2rem;
```

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -97,45 +101,45 @@ html {
-  color: white;   
-  text-shadow: 1px 1px rgba(0, 0, 0, 0.5); 
+  font-weight: 100;
+  color: white;
```
Prefer HEX colors over text ones
```suggestion
  color: #fff;
```

ArtemG22 replied:
fixed

ArtemG22 committed:
refactor: remove extra margins in the html file

ArtemG22 committed:
refactor: correct the case of the title and buttons

ArtemG22 committed:
refactor: fix color value in css properties

ArtemG22 committed:
refactor: eliminate unnecessary margins and spaces in css properties

ArtemG22 commented:
Current assembly by link http://restaurant111022.surge.sh/ Did not close only two punks. 1. Fixed size of the section (which Yura spoke about). 2. Change the logo format from PNG to SVG format.

ArtemG22 committed:
refactor: replace the logo format from png to svg

ArtemG22 committed:
refactor: fix margins in the navigation bar

ArtemG22 commented:
Replaced the logo format with SVG. Was everything done correctly? http://restaurant1110223.surge.sh/

ArtemG22 committed:
refactor: replace the logo format from svg to text

magicsd commented on index.html:
```diff
@@ -15,50 +15,7 @@
-      <svg class="logo">
-        <use xlink:href="#logo"></use>
-      </svg>
+     <a class="logo" hrev="#">Lambda</a>
```
```suggestion
     <a class="logo" href="#">Lambda</a>
```

ArtemG22 replied:
fixed

magicsd replied:
If so, where's the commit with the fix? 😄

ArtemG22 replied:
corrected

ArtemG22 committed:
refactor: fix an error in writing a function in the logo container

ArtemG22 committed:
refactor: delete social network icons in png format

ArtemG22 committed:
feat: add social network icons and connection scripts

ArtemG22 committed:
refactor: change the upper indentation of the promo-decoration container

ArtemG22 commented:
http://restaurant181022.surge.sh/

ArtemG22 committed:
refactor: add the text-decoration-line function to the menu bar in hover mode

ArtemG22 commented:
https://restaurant181022-2.surge.sh/

ArtemG22 committed:
refactor: delete the text-decoration-line property of the menu block in hover mode

magicsd commented:
No need to add comment new staging link every time, just update it in the PR description

ArtemG22 commented:
I got you

ArtemG22 committed:
refactor: add an underscore above the active menu item

ArtemG22 committed:
refactor: replace the menu line indentation parameter from percent to rem

ArtemG22 committed:
refactor: remove an extra navigation panel selector

ArtemG22 commented:
Can I merge branches now?

ynworld commented:
> 

The page still doesn't have a fixed width.
![Screen Shot 2022-10-24 at 09 11 40](https://user-images.githubusercontent.com/109351999/197534093-d2c0c2cf-b515-4d86-9482-e384e7d50709.png)

![Screen Shot 2022-10-24 at 09 11 58](https://user-images.githubusercontent.com/109351999/197533811-2cea40c8-5d1e-4fc1-990f-c707abc465e8.png)
g)

ynworld commented:
Home still has a hover effect.

![Screen Shot 2022-10-24 at 09 13 55](https://user-images.githubusercontent.com/109351999/197534172-cee07d6d-0f1c-4779-ad0e-a95e7689f938.png)

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,220 @@
+}
+
+html {
+  font-size: 100%;
```
Use knowledge from Jonas's course
```suggestion
  font-size: 62.5%;
```

ArtemG22 committed:
refactor: apply font size as a percentage for the entire project

ArtemG22 committed:
refactor: add a promo container and define the layout size

ArtemG22 committed:
refactor: fix property values ​​for promo-slogan and slogan selector

ArtemG22 committed:
refactor: fix property values ​​for promo-decoration and decoration selectors

ArtemG22 committed:
refactor: fix button property values ​​in promo section

ArtemG22 committed:
refactor: fix menu bar property values

ArtemG22 commented:
keep working on bugs

ArtemG22 committed:
refactor: fix the values of the menu bar properties

ArtemG22 commented:
keep working on bugs

ArtemG22 committed:
refactor: remove unnecessary css properties of the menu block

ArtemG22 committed:
refactor: center the layout in the browser

ArtemG22 committed:
refactor: delete the link to the Home item

ArtemG22 committed:
refactor: delete an extra container in Header

ArtemG22 committed:
refactor: delete invalid flex property

ArtemG22 committed:
refactor: remove unnecessary menu bar properties

magicsd commented on index.html:
```diff
@@ -17,30 +17,28 @@
         </ul>
       </div>
+      <ul class="social-links">
+        <li><a class="social-link" href="#">
+            <ion-icon class="social-icon" name="logo-twitter"></ion-icon>
+          </a></li>
```
```suggestion
        <li>
          <a class="social-link" href="https://twitter.com" target="_blank">
            <ion-icon class="social-icon" name="logo-twitter"></ion-icon>
          </a>
        </li>
```

ArtemG22 replied:
Corrected

magicsd replied:
Formatting still crooked

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -81,7 +81,7 @@ body {
   color: #fff;
   font-style: normal;
-  font-size: 2.1rem;
+  font-size: 2.14rem;
```
It'd be weird for designer to pick font size of `21.4px` 😅

ArtemG22 replied:
corrected

ArtemG22 committed:
refactor: specify the URLs of social resources

ArtemG22 committed:
refactor: fix font sizes and menu bar margins

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+
+html {
+  font-size: 62.5%;
+  overflow-x: hidden;
```
```suggestion
```

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+
+body {
+  line-height: 1;
+  overflow-x: hidden;
```
```suggestion
```

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+}
+
+body {
+  line-height: 1;
```
🤔

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+  font-size: 9rem;
+  line-height: 11.4rem;
+  font-weight: 100;
+  color: #F5F5F5;
```
Let's use lowercased variant for all colors
```suggestion
  color: #f5f5f5;
```

ArtemG22 replied:
ok

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+  overflow-x: hidden;
+}
+
+.promo {
+  max-width: 160rem;
+  margin: 0 auto;
+}
```
```suggestion
.wrapper {
  width: 160rem;
  margin: 0 auto;
}
```
And put `promo` container inside `wrapper` container

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+.header {
+  position: absolute;
+  display: flex;
+  width: 160rem;
+  height: 12.2rem;
```
If our parent container is 160rem, then
```suggestion
  width: 100%;
  height: 12rem;
```

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+  height: 12.2rem;
+  justify-content: center;
+  align-items: center;
+  padding: 0 16.8rem;
```
Looks like we don't need to set vertical padding to `0`
```suggestion
  padding-left: 18rem;
  padding-right: 18rem;
```

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+}
+
+.logo-menu-bar {
+  height: inherit;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  width: inherit;
```
I bet it's redundant
```suggestion
  display: flex;
  justify-content: space-between;
  align-items: center;
```

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+  text-transform: uppercase;
+  font-size: 3.6rem;
+  opacity: 0.9;
+  padding: 1rem;
```
Redundant
```suggestion
```

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+  font-style: normal;
+  font-size: 2.2rem;
+  font-weight: 600;
+  transition: all 0.3;
```
```suggestion
  transition: all 0.3s;
```

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+  border-top: 0.38rem solid transparent;
+}
+
+.main-nav-link:hover,
+.main-nav-link:active {
+  opacity: 75%;
+  transition: 0.3s;
+}
+
+.main-nav-link.is-active {
+  opacity: 100%;
+  border-top: 0.38rem solid #fff;
+}
+
+.main-nav-link:focus {
+  opacity: 84%;
+  border-top: 0.38rem solid #fff;
+}
```
Should be revalidated, strange logic with opacities

ArtemG22 replied:
fixed

magicsd replied:
Not fixed

ArtemG22 replied:
I continue to analyze this moment

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+}
+
+.social-link:hover,
+.social-link:active {
```
Why active is also here?

ArtemG22 replied:
fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,230 @@
+.social-link:hover,
+.social-link:active {
+  opacity: 50%;
+  transition: 0.3s;
```
```suggestion
  transition: opacity 0.3s;
```

ArtemG22 replied:
fixed

magicsd replied:
Not fixed

ArtemG22 replied:
I continue to analyze this moment

ArtemG22 committed:
refactor: put the entire site in a new container and assign properties to it

ArtemG22 committed:
refactor: change the size of promo section objects

ArtemG22 committed:
refactor: correct incorrect values of promo section properties

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,225 @@
+
+.promo-section {
+  width: 160rem;
+  height: 75rem;
```
Section height should be defined automatically

ArtemG22 replied:
Fixed

magicsd replied:
Why do you comment "fixed" when there're no any changes applied?

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,225 @@
+.promo-banner {
+  background: url('../gallery/banners/promo-banner.jpg');
+  background-size: cover;
+  width: 160rem;
+  height: 75rem;
```
Look redundant
```suggestion
```

ArtemG22 replied:
Excluding the width property from the banner properties went fine. But if I exclude the height parameter, then the banner is clipped.
![2022-12-21_17-48-46](https://user-images.githubusercontent.com/110885647/208943876-4f5372b8-ca26-421c-8728-a863f024d999.png)

magicsd replied:
You have to fix this
<img width="165" alt="image" src="https://user-images.githubusercontent.com/34685713/209054605-b7ad0ee5-74d4-4896-9d3b-2cf0bcb903f1.png">

magicsd replied:
And please remove all unnecessary styles (keep only those that make sense and really meaningful for the layout)

ArtemG22 replied:
> You have to fix this <img alt="image" width="165" src="https://user-images.githubusercontent.com/34685713/209054605-b7ad0ee5-74d4-4896-9d3b-2cf0bcb903f1.png">

Fixed

ArtemG22 committed:
refactor: build the correct structure of the block of social networks

ArtemG22 commented:
> > 
> 
> The page still doesn't have a fixed width. ![Screen Shot 2022-10-24 at 09 11 40](https://user-images.githubusercontent.com/109351999/197534093-d2c0c2cf-b515-4d86-9482-e384e7d50709.png)
> 
> ![Screen Shot 2022-10-24 at 09 11 58](https://user-images.githubusercontent.com/109351999/197533811-2cea40c8-5d1e-4fc1-990f-c707abc465e8.png) g)

Fixed

ArtemG22 committed:
refactor: change the properties of the logo menu block

ArtemG22 committed:
refactor: Replace promo section height property

ArtemG22 committed:
refactor: Removing the width parameter from the promo section banner

ArtemG22 committed:
refactor:
remove an extra height parameter in the properties of the main block with a slogan and other elements

ArtemG22 committed:
refactor: replace class name and main block selector in promo section

magicsd commented on css/style.css:
```diff
@@ -134,14 +134,12 @@ html {
 .promo-slogan {
   width: inherit;
-  height: 25rem;
+  height: 53.4rem;
```
Remove `height`, it should be defined automatically

ArtemG22 replied:
Fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,220 @@
+.main-nav-link:hover,
+.main-nav-link:active {
+  opacity: 75%;
+  transition: 0.3s;
```
`transition` of what? You have to specify it
```suggestion
  transition: opacity 0.3s;
```

ArtemG22 replied:
Fixed. Hope it's right.

magicsd replied:
The suggested change is right, so you can be sure 😄

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,220 @@
+  color: #fff;
+  text-transform: uppercase;
+  font-size: 3.6rem;
+  opacity: 0.9;
```
Why everywhere `opacity` in percentage but not it's not?

ArtemG22 replied:
Fixed. Did not notice the loss of the sign "%".

magicsd commented on index.html:
```diff
@@ -1,14 +1,70 @@
 <body>
-    <h1>LAMBDA restaurant</h1>
-    
+  <div class="wrapper">
```
The structure is expected to be something like this
```html
<div class="main-wrapper">
  <main>
    <section class="section-promo">
      <header class="promo-header">
         <img />
         <nav>
           <ul>...</ul>
         </nav>
         <ul>...</ul>
       </header>
      <div class="promo-body">...</div>
    </section>
  </main>
</div>
```

ArtemG22 replied:
Fixed

magicsd replied:
Not really. You put `<div class="logo-menu-bar">` which is redundant

ArtemG22 replied:
Corrected. Not sure I went the right way with regards to spacing between the logo and the menu.

magicsd replied:
Use `flex-box`

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,220 @@
+  gap: 0.8rem;
+  padding-right: 2.8rem;
+  align-items: center;
+  
```
```suggestion
```

ArtemG22 replied:
Fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,220 @@
+/******************************/
+
+.promo-section {
+  width: 160rem;  
```
Main wrapper will do the job
```suggestion
```

ArtemG22 replied:
Corrected. Understood. Disabled this selector due to unnecessary

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,220 @@
+}
+
+.promo-slogan {
+  width: inherit;  
```
It's inherited automatically, here and everywhere

ArtemG22 replied:
Fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,220 @@
+
+.buttons-bar {
+  width: inherit;
+  height: 5.6rem;
```
Mostly you have to specify height for elements not for containers. Container's height is usually defined by height of its elements.
```suggestion
```

ArtemG22 replied:
Corrected. Remembered for the future

ArtemG22 committed:
refactor: remove height parameter from css file and wrong values

ArtemG22 committed:
refactor: fix html file structure for promo section

ArtemG22 committed:
refactor: fix navbar properties

ArtemG22 commented:
Fixed a part of the errors, updated the deployment. I keep working on the bugs.

ArtemG22 committed:
refactor: remove disabled properties and empty selectors

ArtemG22 committed:
refactor: Remove container height parameter with buttons - promo section

ArtemG22 commented:
A few more old remarks remain. Tomorrow I will continue to deal with them

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,215 @@
+  gap: 0.8rem;
+}
+
+.header-shadow {
+  width: 160rem;
+  position: absolute;  
+  height: 12rem;    
+  background: linear-gradient(to bottom, black -100%, transparent 90%);  
+}
```
Let's rename `promo-header` to `main-header` because it's doesn't relate to section "promo" but to whole page. Therefore:
```suggestion
.main-header:before {
  content: '',
  width: 100%;
  height: 100%;
  top: 0,
  left: 0,
  position: absolute;  
  background: linear-gradient(to bottom, black -100%, transparent 90%);  
}
```

ArtemG22 replied:
I understood the principle of the pseudo-element. applied

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,215 @@
+  position: absolute;
+  display: flex;  
+  height: 12rem;  
+  align-items: center;
```
Are you sure you need it here?

ArtemG22 replied:
I decide this moment. When these properties are excluded, the template breaks.

magicsd replied:
Refresh `flex` knowledge https://css-tricks.com/snippets/css/a-guide-to-flexbox/ (`self-start`, `self-end`)

ArtemG22 replied:
ok. I'll do it right now

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,215 @@
+  background: linear-gradient(to bottom, black -100%, transparent 90%);  
+}
+
+.logo-menu-bar {
+  display: flex;
+  justify-content: space-between;
+  gap: 22.8rem;
+  align-items: center;
+}
```
Redundant

ArtemG22 replied:
deleted. Changed the properties of the logo. Adding padding on the right side

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,215 @@
+/* PROMO SECTION */
+/******************************/
+
+.promo-banner {
```
It looks like whole section "promo" has this background
```suggestion
.promo-section {
```

ArtemG22 replied:
Fixed

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,215 @@
+.promo-banner {
+  background: url('../gallery/banners/promo-banner.jpg');
+  background-size: cover;  
+  padding-top: 12rem;  
```
Looks useless
```suggestion
```

ArtemG22 replied:
Looking for a solution. If you just remove the top padding, you get the following
![2022-12-26_10-46-50](https://user-images.githubusercontent.com/110885647/209520575-174ecca0-05c5-4080-97bc-bbfe80420fd8.png)

ArtemG22 committed:
refactor: Remove extra class logo-menu-bar and its properties

magicsd commented on css/style.css:
```diff
@@ -55,6 +48,7 @@ html {
   text-transform: uppercase;
   font-size: 3.6rem;
   opacity: 90%;
+  margin-right: 22rem;
```
Consider `margin-right: auto;`

ArtemG22 replied:
did not solve the problem
![2022-12-26_10-53-19](https://user-images.githubusercontent.com/110885647/209521211-bbb0a1e1-95d3-46e6-87d5-9b5bc06d805c.png)

ArtemG22 replied:
I will look for a solution

ArtemG22 committed:
refactor: Changing the structure of the html file separating the header section from other sections

ArtemG22 committed:
refactor: remove an extra class in the site structure responsible for the background

ArtemG22 committed:
refactor: add a height property for the back photo and a width property for the header

ArtemG22 commented:
Added height property for background and width property for header. I am looking for a solution to replace or remove these values.

ArtemG22 committed:
refactor: Add height property and value to main-header

ArtemG22 committed:
refactor: remove extra paddings in the main menu block

ArtemG22 committed:
refactor: set blocks in the site header using the justify-content function

ArtemG22 committed:
refactor: set hover property for menu items

ArtemG22 committed:
refactor: remove extra hover property from active menu item pseudo-element

ynworld commented on css/style.css:
```diff
@@ -67,45 +67,55 @@ html {
-.main-nav-link,
-.main-nav-link:link,
-.main-nav-link:visited,
-.main-nav-link:active {
```
Why did you remove the :link, :visited, and :active pseudo-classes?

ArtemG22 replied:
I decided that adding them at this stage - I am ahead of the curve. Is it worth returning these properties?

ynworld replied:
It's standard to have them for links even if all the styles are the same.

magicsd replied:
Let's finish the most important part with layout and positioning and then tune links as a next step

ArtemG22 committed:
refactor: add pseudo-classes for the menu block

ArtemG22 committed:
refactor: make changes according to the comments

ArtemG22 committed:
refactor: add properties for the menu block

ArtemG22 committed:
refactor: add transition property to pseudo menu items and set size of social media icons

ArtemG22 committed:
refactor: remove superfluous properties of menu block pseudo elements

ArtemG22 committed:
refactor: remove redundant navbar properties

ArtemG22 committed:
refactor: add spacing between items in the navbar

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,233 @@
+  align-items: center;
+}
+
+.social-link {
+  display: flex;
+  width: 2rem;
+  height: 2rem;
+}
```
Let's make container 24px x 24px
```suggestion
.social-link {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 2.4rem;
  height: 2.4rem;
}
```

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,233 @@
+  transition: opacity 0.3s;
+}
+
+.social-icon {
+  width: 2rem;
+  height: 2rem;
+}
```
It should work for icons since it's just a svg
```suggestion
.social-icon {
  font-size: 2rem;
}
```

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,233 @@
+  height: 75rem;
+}
+
+.promo-slogan-and-elements {
+  margin-top: 9.6rem;
+}
```
Looks redundant, we can safely remove this style and container and move this margin to `.promo-slogan`

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,233 @@
+  text-align: center;
+  font-size: 9rem;
+  line-height: 11.4rem;
+  font-weight: 100;
```
Looks strange

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,233 @@
+.promo-slogan {
+  font-family: "Yeseva One", sans-serif;
+  text-transform: lowercase;
+  text-align: center;
```
I think we can move it to `.promo-section`

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,233 @@
+  text-shadow: 0.1rem 0.1rem rgba(0, 0, 0, 0.5);
+}
+
+.promo-decoration {
+  text-align: center;
+  margin-top: 2.8rem;
+}
+
+.decoration {
+  width: 23.4rem;
+}
```
Looks like we can remove container and leave only `img` with this class
```suggestion
.promo-decoration {
  margin-top: 2.8rem;
  width: 23.4rem;
}
```

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,233 @@
+  width: 23.4rem;
+}
+
+.buttons-bar {
+  padding: 0 2rem;
+  text-align: center;
+  margin-top: 2.8rem;
+}
```
Looks redundant as well as div

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,233 @@
+  margin-top: 2.8rem;
+}
+
+.buttons {
+  font-family: "Arial", sans-serif;
+  text-transform: uppercase;
+}
```
Button styles like font-family and text-transform should belong to specific button styling. And it's just a container
```suggestion
.promo-buttons {
  display: flex;
  justify-content: center;
  gap: 2.8rem;
  margin-top: 2.8rem;
}
```
it's not really correct to name it `buttons` because it's too common, we'll have more sections and this block looks promo section specific

ArtemG22 committed:
refactor: remove unnecessary elements and their properties from the promo-section block

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,219 @@
+  text-decoration: none;
+  font-size: 1.8rem;
+  font-weight: 100;
+  padding: 1.6rem 4.8rem;
```
It's better to give only horizontal paddings and set specific height, therefore all the buttons will be of the same predictable height all the time.

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,219 @@
+  padding: 1.6rem 4.8rem;
+  border: none;
+  cursor: pointer;
+  font-family: inherit;
```
?

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,219 @@
+  display: inline-block;
+  text-decoration: none;
+  font-size: 1.8rem;
+  font-weight: 100;
```
?

magicsd commented on css/style.css:
```diff
@@ -0,0 +1,219 @@
+  justify-content: center;
+  gap: 2.8rem;
+  margin-top: 2.8rem;
+  font-family: "Arial", sans-serif;
+  text-transform: uppercase;
```
Must be moved to button styles

ArtemG22 committed:
refactor: remove unnecessary properties of the buttons block

ArtemG22 committed:
refactor: remove extra padding and place promo section elements in flex container

magicsd commented on index.html:
```diff
@@ -1,14 +1,74 @@
+      <nav>
+        <ul class="nav">
+          <li>
+            <a class="nav-link is-active">Home</a>
```
`a` tag must always has `href` attribute

magicsd commented on index.html:
```diff
@@ -1,14 +1,74 @@
+    </header>
+    <main>
+      <section class="promo-section">
+        <p class="promo-slogan">The right ingredients<br>for the right food</p>
```
Since it's a heading, some `h` tag should be used.
```suggestion
        <h2 class="promo-slogan">The right ingredients<br>for the right food</h2>
```
`h2` because `h1` tag has meaning for search engines and should be used to match such kind of goals

magicsd commented on index.html:
```diff
@@ -1,14 +1,74 @@
-    
+  <div class="main-wrapper">
+    <header class="main-header">
+      <a class="logo" href="#">Lambda</a>
```
```suggestion
      <a href="/">
        <h1 class="logo">Lambda</h1>
      </a>
```
See description why in another comment.
Link should pass us to the root I guess.

ArtemG22 committed:
refactor: add title tags and optimize images

ArtemG22 committed:
refactor: add additional logo container and remove invalid properties

ArtemG22 committed:
refactor: remove invalid logo block property

ArtemG22 committed:
refactor: remove invalid h1 and h2 heading properties
